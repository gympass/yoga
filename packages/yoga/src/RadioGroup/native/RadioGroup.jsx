import React, { useState } from 'react';
import { View } from 'react-native';
import { string, func, oneOfType, number, bool, node } from 'prop-types';
import styled from 'styled-components';

import RadioGroupContext from '../RadioGroupContext';
import RadioButton from './Button/RadioButton';

const Group = styled.View(
  () => `
    align-self: center;
    position: relative;
    flex-direction: row;
  `,
);

const GrayLine = styled.View(
  ({
    theme: {
      yoga: {
        components: { radiogroup },
      },
    },
  }) => `
    position: absolute;
    height: 100%;
    border-radius: ${radiogroup.button.border.radius}px;
    border-width: ${radiogroup.button.border.width}px;
    border-color: ${radiogroup.button.border.color};
  `,
);

const RadioGroup = ({
  onChange = () => {},
  selectedValue = '',
  small = false,
  children,
  ...rest
}) => {
  const [groupSize, setGroupSize] = useState(0);

  const isButton = React.Children.map(children, child => child.type).every(
    child => child === RadioButton,
  );

  return (
    <RadioGroupContext.Provider
      value={{
        selectedValue,
        onChange,
        small,
        ...rest,
      }}
    >
      <View>
        <Group
          {...rest}
          onLayout={({
            nativeEvent: {
              layout: { width },
            },
          }) => setGroupSize(width)}
        >
          {Boolean(isButton && groupSize) && (
            <GrayLine style={{ width: groupSize }} />
          )}

          {children}
        </Group>
      </View>
    </RadioGroupContext.Provider>
  );
};

RadioGroup.propTypes = {
  onChange: func,
  selectedValue: oneOfType([string, number]),
  small: bool,
  children: node.isRequired,
};

export default RadioGroup;
