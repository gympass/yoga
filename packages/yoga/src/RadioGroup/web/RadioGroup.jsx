import React from 'react';
import { string, func, oneOfType, number, bool, node, shape } from 'prop-types';
import styled from 'styled-components';

import RadioGroupContext from '../RadioGroupContext';

const Group = styled.div.attrs({
  role: 'radiogroup',
})`
  display: flex;

  width: min-content;

  ${({ full }) => `
    ${full ? `width: 100%;` : ``}
  `}
`;

const RadioGroup = ({
  name = '',
  onChange = () => {},
  selectedValue = '',
  small = false,
  full = false,
  children = null,
  style = {},
  ...rest
}) => (
  <RadioGroupContext.Provider
    value={{
      selectedValue,
      name,
      onChange,
      small,
      ...rest,
    }}
  >
    <Group full={full} style={style} {...rest}>
      {children}
    </Group>
  </RadioGroupContext.Provider>
);

RadioGroup.propTypes = {
  name: string,
  onChange: func,
  selectedValue: oneOfType([string, number]),
  small: bool,
  full: bool,
  children: node,
  style: shape({}),
};

export default RadioGroup;
