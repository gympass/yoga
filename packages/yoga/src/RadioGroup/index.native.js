import { RadioGroup, RadioButton, Radio } from './native';

const ExportRadioGroup = Object.assign(RadioGroup, {
  Button: RadioButton,
  Radio,
});

export default ExportRadioGroup;
