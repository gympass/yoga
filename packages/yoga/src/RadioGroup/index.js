import { RadioGroup, RadioButton, Radio } from './web';

const ExportRadioGroup = Object.assign(RadioGroup, {
  Button: RadioButton,
  Radio,
});

export default ExportRadioGroup;
