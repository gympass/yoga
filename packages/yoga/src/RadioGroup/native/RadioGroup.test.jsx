import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';

import RadioGroup from '..';
import ThemeProvider from '../../Theme';

describe('<RadioGroup />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with radio button as children', () => {
      const { container, getAllByText, toJSON } = render(
        <ThemeProvider>
          <RadioGroup>
            <RadioGroup.Button>Option 1</RadioGroup.Button>
            <RadioGroup.Button>Option 2</RadioGroup.Button>
            <RadioGroup.Button>Option 3</RadioGroup.Button>
          </RadioGroup>
        </ThemeProvider>,
      );

      expect(toJSON(container)).toMatchSnapshot();
      expect(getAllByText(/^Option/)).toHaveLength(3);
    });

    it('should match snapshot with small radios', () => {
      const { container, toJSON } = render(
        <ThemeProvider>
          <RadioGroup small>
            <RadioGroup.Button>Option 1</RadioGroup.Button>
            <RadioGroup.Button>Option 2</RadioGroup.Button>
            <RadioGroup.Button>Option 3</RadioGroup.Button>
          </RadioGroup>
        </ThemeProvider>,
      );

      expect(toJSON(container)).toMatchSnapshot();
    });

    it('should has setted the selected value', () => {
      const { toJSON, getByText } = render(
        <ThemeProvider>
          <RadioGroup selectedValue="Option 2">
            <RadioGroup.Button>Option 1</RadioGroup.Button>
            <RadioGroup.Button>Option 2</RadioGroup.Button>
            <RadioGroup.Button>Option 3</RadioGroup.Button>
          </RadioGroup>
        </ThemeProvider>,
      );

      expect(toJSON()).toMatchSnapshot();
      expect(getByText('Option 2').props.checked).toBe(true);
    });
  });
});

describe('Events', () => {
  it('should set the onchange for its children', () => {
    const onChangeMock = jest.fn();

    const { getByText } = render(
      <ThemeProvider>
        <RadioGroup onChange={onChangeMock}>
          <RadioGroup.Button>Option 1</RadioGroup.Button>
          <RadioGroup.Button>Option 2</RadioGroup.Button>
          <RadioGroup.Button>Option 3</RadioGroup.Button>
        </RadioGroup>
      </ThemeProvider>,
    );

    const firstOption = getByText('Option 1');

    fireEvent.press(firstOption);

    expect(onChangeMock).toHaveBeenCalled();
  });
});
