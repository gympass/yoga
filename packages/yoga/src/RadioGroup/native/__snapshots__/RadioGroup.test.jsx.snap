// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> Snapshots should has setted the selected value 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "position": "relative",
        },
      ]
    }
  >
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 40,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 1
      </Text>
    </View>
    <View
      accessible={true}
      checked={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#D8385E",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 40,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={true}
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Rubik",
              "fontWeight": "700",
            },
          ]
        }
      >
        Option 2
      </Text>
    </View>
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 40,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 3
      </Text>
    </View>
  </View>
</View>
`;

exports[`<RadioGroup /> Snapshots should match snapshot with radio button as children 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "position": "relative",
        },
      ]
    }
  >
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 40,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 1
      </Text>
    </View>
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 40,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 2
      </Text>
    </View>
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={false}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 40,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 3
      </Text>
    </View>
  </View>
</View>
`;

exports[`<RadioGroup /> Snapshots should match snapshot with small radios 1`] = `
<View>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignSelf": "center",
          "flexDirection": "row",
          "position": "relative",
        },
      ]
    }
  >
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 1
      </Text>
    </View>
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 2
      </Text>
    </View>
    <View
      accessible={true}
      checked={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      small={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderRadius": 9999,
            "fontSize": 14,
            "height": 32,
            "justifyContent": "center",
            "paddingLeft": 24,
            "paddingRight": 24,
          },
        ]
      }
    >
      <Text
        checked={false}
        style={
          Array [
            Object {
              "color": "#231B22",
              "fontFamily": "Rubik",
            },
          ]
        }
      >
        Option 3
      </Text>
    </View>
  </View>
</View>
`;
